tests/integration/test_logcollector/test_age/test_age_basic.py

```diff
@@ -0,0 +1,151 @@
+    {'location': folder_path_regex, 'log_format': 'syslog', 'age': '200d'},
+]
+
+configurations = load_wazuh_configurations(configurations_path, __name__, params=parameters, metadata=metadata)
+configuration_ids = [f"{x['LOCATION'], x['LOG_FORMAT'], x['AGE']}" for x in parameters]
+
+
+@pytest.fixture(scope="module", params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope="function")
+def get_files_list():
+    """Get file list to create from the module."""
+    return file_structure
+
+
+@pytest.fixture(scope="module")
+def get_local_internal_options():
+    """Get local internal options from the module."""
+    return local_internal_options
+
+
+@pytest.mark.xfail(reason="Expected error. Issue https://github.com/wazuh/wazuh/issues/8438")
+def test_configuration_age_basic(get_local_internal_options, configure_local_internal_options, get_files_list,
+                                 create_file_structure, get_configuration, configure_environment, restart_logcollector):
+    """Check if logcollector works correctly and uses the specified age value.
+
+    Check that those files that have not been modified for a time greater than age value, are ignored for logcollector.
+    Otherwise, files should not be ignored. Also, it checks logcollector detect modification time changes in monitored
+    files and catch new logs from ignored and not ignored files.
+
+    Raises:
+        TimeoutError: If the expected callbacks are not generated.
+    """
+
+    cfg = get_configuration['metadata']
+    age_seconds = time_to_seconds(cfg['age'])
+
+    for file in file_structure:
+        absolute_file_path = os.path.join(file['folder_path'], file['filename'])
+        wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+        log_callback = logcollector.callback_file_matches_pattern(cfg['location'], absolute_file_path, prefix=prefix)
+        wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                                error_message=f'{file["filename"]} was not detected')
+
+        if int(age_seconds) <= int(file['age']):
+            log_callback = logcollector.callback_ignoring_file(
+                absolute_file_path, prefix=prefix)
+            wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                                    error_message=f'{file["filename"]} was not ignored')
+
+        else:
+            with pytest.raises(TimeoutError):
+                log_callback = logcollector.callback_ignoring_file(absolute_file_path,
+                                                                   prefix=prefix)
+                wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                                        error_message=f'{file["filename"]} was not ignored')
+
+    for file in file_structure:
+        with open(absolute_file_path, 'a') as file_to_write:
```

Comment:
I think this variable should be redefined. Other wise, the value of `absolute_file_path` will be the one from the last iteration of the previous loop.